src/uwtools/cli.py

```diff
@@ -305,6 +309,132 @@ def _dispatch_forecast_run(args: Namespace) -> bool:
+        partial(_check_file_vs_format, STR.outfile, STR.outfmt),
+    ]
+
+
+def _add_subparser_rocoto_validate(subparsers: Subparsers) -> SubmodeChecks:
+    """
+    Subparser for mode: rocoto validate
+
+    :param subparsers: Parent parser's subparsers, to add this subparser to.
+    """
+    parser = _add_subparser(subparsers, STR.validate, "Validate Rocoto XML")
+    required = parser.add_argument_group(TITLE_REQ_ARG)
+    _add_arg_input_file(required)
+    optional = _basic_setup(parser)
+    _add_arg_input_format(optional, choices=[FORMAT.rocoto])
+    checks = _add_args_quiet_and_verbose(optional)
+    return checks + [
+        partial(_check_file_vs_format, STR.infile, STR.infmt),
+    ]
+
+
+def _dispatch_rocoto(args: Namespace) -> bool:
+    """
+    Dispatch logic for rocoto mode.
+
+    :param args: Parsed command-line args.
+    """
+    return {
+        STR.realize: _dispatch_rocoto_realize,
+        STR.validate: _dispatch_rocoto_validate,
+    }[
+        args.submode
+    ](args)
+
+
+def _dispatch_rocoto_realize(args: Namespace) -> bool:
+    """
+    Dispatch logic for rocoto realize submode. Validate input and output.
+
+    :param args: Parsed command-line args.
+    """
+    with resources.as_file(resources.files("uwtools.resources")) as resc:
+        input_template = resc / "rocoto.jinja2"
+        output_template = resc / "rocoto.jsonschema"
+
+    success = True
+    if args.input_format == FORMAT.yaml and args.output_format == FORMAT.rocoto:
+        if uwtools.config.validator.validate_yaml(
+            config_file=args.input_file, schema_file=input_template
+        ):
+            uwtools.rocoto.write_rocoto_xml(
+                input_yaml=args.input_file,
+                input_template=str(input_template),
+                rendered_output=str(args.output_file),
+            )
+            success = uwtools.rocoto.validate_rocoto_xml(
+                input_xml=args.output_file, schema_file=str(output_template)
+            )
+        else:
+            success = False
+    else:
+        success = False
+    return success
+
+
+def _dispatch_rocoto_validate(args: Namespace) -> bool:
+    """
+    Dispatch logic for rocoto validate submode.
+
+    :param args: Parsed command-line args.
+    """
+    with resources.as_file(resources.files("uwtools.resources")) as resc:
+        schema_file = resc / "rocoto.jsonschema"
+
+    success = True
+    if args.input_format == FORMAT.rocoto:
+        success = uwtools.rocoto.validate_rocoto_xml(
+            input_xml=args.input_file, schema_file=str(schema_file)
+        )
+    else:
+        success = False
+    if args.verbose: #pragma: no cover
+        errors = str(etree.RelaxNG.error_log).split("\n")
+        log_method = logging.error if len(errors) else logging.info
+        log_method(
+            "%s Rocoto validation error%s found", len(errors), "" if len(errors) == 1 else "s"
+        )
+        for line in errors:
+            logging.error(line)
```

Comment:
All of this should be logged directly from the `validate_rocoto_xml` method.

Reply:
This was another debate point for me, whether to handle the errors here, offload to another method in `cli()`, or to expand `rocoto()` to accept args - expanding `rocoto()` is probably the way to go in the end

Reply:
correction: not args, still thinking in old terms, got it!